### PR TITLE
QEA-1075: Gridium Unit Tests - Refactor

### PR DESCRIPTION
**Pull request by sethuster**
Get Jenkins to run the tests and push the gem to the repo if successful.  

This change requires that the S3 settings be provided to the Gridium container somehow.  The following environment variables must be present in the Gridium container for the s3 tests to complete:

S3_ACCESS_KEY_ID=
S3_DEFAULT_REGION=
S3_ROOT_BUCKET=
S3_SECRET_ACCESS_KEY=

Values must be valid! Do not commit them to github. :)

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+version: '2'
+services:
+  gridium:
+    volumes:
+      - $GRIDIUMPATH:/gridium
```
Comment (yet-another-lucas) [inline docker-compose.dev.yml]: Yes. Huge quality of life improvement right here.

**Comment by yet-another-lucas** [comment]
When we tried to run all the specs, we encountered errors that don't appear when running individual files:
```
#from inside gridium image
rspec -P *_spec.rb
```

File: spec/element_spec.rb [hunk 2]
```diff
@@ -13,9 +13,6 @@
     Gridium.config.browser = :firefox
   end
 
-  after :all do
-    Gridium.config.browser_source = :local
-  end
```
Comment (yet-another-lucas) [inline spec/element_spec.rb]: Ahh, was I polluting the state for subsequent tests here?

File: lib/s3.rb [hunk 3]
```diff
@@ -1,14 +1,14 @@
 require 'aws-sdk'
 module Gridium
 
-
     class GridiumS3
 
         DELIMITER = "/"
 
         def initialize(project_name, subdirectory_name='screenshots')
+            puts "*** REGION: #{ENV['S3_DEFAULT_REGION']}"
```
Comment (yet-another-lucas) [inline lib/s3.rb]: Just a nitpick. This is shows up a lot, maybe it can be debug level.

File: spec/element_spec.rb (hunk 2 above)
Comment (sethuster) [inline spec/element_spec.rb]: heh - yup.

**Pull request merged**